<a href="https://colab.research.google.com/github/tanpaterusan/data-science-2026/blob/main/Pertemuan2_Amalia_240401010284.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# 1 Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

In [ ]:
# 2 Eksplorasi Awal
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Kuis 1:
Berapa jumlah baris dan kolom dataset Titanic?

> 891 baris dan 12 kolom.




Kuis 2: Kolom mana yang memiliki nilai kosong (missing)?


> Age sebanyak 177 baris, Cabin sebanyak 687 baris dan Embarked sebanyak 2 baris.



In [ ]:

# 3 Filtering — Penumpang Wanita Kelas 1
wk1 = df[(df['Sex']=='female') &
         (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.9680851063829787 34.61176470588235


Kuis 3: Rata-rata usia penumpang wanita kelas 1 yang selamat?


> 34 tahun.





In [ ]:
# 4 Groupby — Tingkat Keselamatan per Kelas
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


Kuis 4: Kelas penumpang mana yang tingkat keselamatan tertinggi?


> Penumpang kelas 3



In [ ]:
# Kuis 5: Tampilkan hasil groupby('Pclass')['Survived'].mean()

print(df.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


**Evaluasi**


1. Apa yang Saya Pelajari
- Eksplorasi Kondisi Data: Cara cek ukuran dataset dan mendeteksi kolom yang bolong (missing values).
- Filtering Multi-Kondisi: Teknik menyaring data spesifik menggunakan operator logika & (contoh: wanita DAN kelas 1).
- Agregasi Kustom: Membuat ringkasan data kustom menggunakan .groupby() dan .agg() untuk menghitung jumlah sekaligus rata-rata bersamaan.


---



2. Temuan Utama Data
- Krisis Data Cabin & Usia: Mayoritas data kabin (Cabin = 687) hilang, dan banyak usia penumpang (Age = 177) tidak tercatat.
- Prioritas Wanita Kelas 1: Penumpang wanita kelas 1 punya peluang selamat yang luar biasa tinggi, mencapai 96.8%.
- Kasta Sosial Menentukan Nyawa: Tingkat keselamatan kelas 1 paling tinggi (63%), diikuti kelas 2 (47.3%), dan kelas 3 paling rendah (24.2%).



---


3. Keterbatasan & Pertanyaan Belajar
- Ada keliru di kunci Jawaban Kuis 4: Kunci jawaban kuis 4 tertulis "Kelas 3", padahal hasil data jelas menunjukkan Kelas 1 yang keselamatannya tertinggi.
- Rata-rata Usia yang Bias: Angka rata-rata usia wanita kelas 1 (34.6 tahun) dihitung tanpa mengisi 177 data usia yang kosong. Apakah hasilnya tetap akurat jika data kosong diisi dulu?
- Abaikan Faktor Finansial: Mengapa kita hanya melihat kelas (Pclass), tapi belum menguji hubungan harga tiket (Fare) langsung dengan peluang selamat?